sync/src/sys/locks/rwlock.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Rust RwLock implementation used in SGX environments
+
+#![allow(dead_code)]
+
+use mc_sgx_tstdc::RwLock as SgxRwLock;
+
+/// SGX rwlock backend to use with the common Rust std lib
+/// [`RwLock`](https://doc.rust-lang.org/std/sync/struct.RwLock.html) interface.
+///
+///
+/// Read locks are implemented using a reference count. This means that clients
+/// are responsible for managing which threads hold the read locks and ensuring
+/// threads only unlock if they are currently holding a reader lock. Write locks
+/// do track which thread owns them.
+pub(crate) struct RwLock {
```

Comment:
Could this be renamed to `ReadWriteLock` ? You can leave it as is If `rw` is intuitive in rust.

Reply:
I think for consistency with https://doc.rust-lang.org/std/sync/struct.RwLock.html and it's backend implementations I'm going to leave this as `RwLock`